### PR TITLE
adds an error message to movables not being removed from the grid

### PR DESCRIPTION
**Pull request by Kylerace**
## About The Pull Request
second times the charm
## Why It's Good For The Game
ive been released from one of my burdens today why not another

### CODE REVIEW

File: code/controllers/subsystem/spatial_gridmap.dm [hunk 1]
```diff
@@ -468,20 +468,34 @@ SUBSYSTEM_DEF(spatial_grid)
...
 
 		if(SPATIAL_GRID_CONTENTS_TYPE_ATMOS)
-			GRID_CELL_REMOVE(intersecting_cell.atmos_contents, old_target)
+			GRID_CELL_REMOVE(intersecting_cell, intersecting_cell.atmos_contents, old_target)
 			SEND_SIGNAL(intersecting_cell, SPATIAL_GRID_CELL_EXITED(exclusive_type), old_target)
 
 	return TRUE
 
+#ifdef UNIT_TESTS
+/mob/living/carbon/human/species/lizard/trolls_the_maintainer
```
Comment (Kylerace) [inline code/controllers/subsystem/spatial_gridmap.dm:486]: will remove

Comment (JohnFulpWillard) [reply code/controllers/subsystem/spatial_gridmap.dm:486]: why

Comment (Kylerace) [reply code/controllers/subsystem/spatial_gridmap.dm:486]: want to fail the tests but the test explorer is annoyingly unresponsive so i test after pushing

Comment (Mothblocks) [reply code/controllers/subsystem/spatial_gridmap.dm:486]: test explorer is junk, `build -DUNIT_TESTS` and then running that works

File: code/controllers/subsystem/spatial_gridmap.dm [hunk 2]
```diff
@@ -490,13 +504,56 @@ SUBSYSTEM_DEF(spatial_grid)
 
 	if(!input_cell)
 		input_cell = get_cell_of(to_remove)
+
+		#ifdef UNIT_TEST
```
Comment (Mothblocks) [inline code/controllers/subsystem/spatial_gridmap.dm:508]: this isnt the right define

Comment (Kylerace) [reply code/controllers/subsystem/spatial_gridmap.dm:508]: :)

File: code/controllers/subsystem/spatial_gridmap.dm [hunk 3]
```diff
@@ -468,20 +468,34 @@ SUBSYSTEM_DEF(spatial_grid)
...
+/mob/living/carbon/human/species/lizard/trolls_the_maintainer/Initialize(mapload)
+	. = ..()
+	become_hearing_sensitive()
+	var/list/datum/spatial_grid_cell/nearby_cells = SSspatial_grid.get_cells_in_range(src, 30)
+
+	for(var/datum/spatial_grid_cell/nearby_cell in nearby_cells)
+		GRID_CELL_ADD(nearby_cell, nearby_cell.hearing_contents, src)
+		GRID_CELL_ADD(nearby_cell, nearby_cell.hearing_contents, src)
+#endif
+
```
Comment (Kylerace) [inline code/controllers/subsystem/spatial_gridmap.dm:498]:
```suggestion
```

File: code/__DEFINES/spatial_gridmap.dm [hunk 4]
```diff
@@ -15,31 +15,59 @@
...
+#define ALL_CHANNELS_OF_CELL(cell) (cell.hearing_contents | cell.client_contents | cell.atmos_contents)
+
 ///whether movable is itself or containing something which should be in one of the spatial grid channels.
 #define HAS_SPATIAL_GRID_CONTENTS(movable) (movable.spatial_grid_key)
 
+#ifdef UNIT_TESTS
+#define UPDATE_GRID_METADATA(movable_or_list, cell) \
+	do { \
+		var/list/_definitely_list = movable_or_list; \
+		if(!istype(movable_or_list, /list)) { \
```
Comment (san7890) [inline code/__DEFINES/spatial_gridmap.dm:27]: can we just use `islist()` here instead

**Pull request closed without merging**